### PR TITLE
Add minimum threshold logic for finding field index holes

### PR DESCRIPTION
**Pull request by lbschanno**
Update the logic for finding field index holes to support minimum thresholds. An index hole is considered to exist for a field/datatype when the count for an index row does not meet the threshold percentage-wise for the count on the corresponding frequency row.

Required to support work for datawave/#825.

### CODE REVIEW

File: src/main/java/datawave/query/util/AllFieldMetadataHelper.java [hunk 1]
```diff
@@ -1061,350 +1059,309 @@ public Set<String> loadDatatypes() throws TableNotFoundException {
...
-     * 
+     *
+     * @param minThreshold
+     *            the minimum percentage threshold required for an index row to be considered NOT a hole on a particular date
      * @return a map of field names and datatype pairs to field index holes
      */
     @Cacheable(value = "getFieldIndexHoles", key = "{#root.target.auths,#root.target.metadataTableName}", cacheManager = "metadataHelperCacheManager")
-    public Map<String,Map<String,FieldIndexHole>> getFieldIndexHoles() throws TableNotFoundException, CharacterCodingException {
-        return getFieldIndexHoles(ColumnFamilyConstants.COLF_I);
+    public Map<String,Map<String,FieldIndexHole>> getFieldIndexHoles(int minThreshold) throws TableNotFoundException, IOException {
```
Comment (ivakegg) [inline src/main/java/datawave/query/util/AllFieldMetadataHelper.java:1068]: I am not sure that an integer percentage is enough granularity.  Can we make this a float/double beween 0.0 and 1.0 ?

Comment (lbschanno) [reply src/main/java/datawave/query/util/AllFieldMetadataHelper.java]: Done, it's now a double that must be between 0.0-1.0.

File: src/main/java/datawave/query/util/AllFieldMetadataHelper.java [hunk 2]
```diff
@@ -1061,350 +1059,307 @@ public Set<String> loadDatatypes() throws TableNotFoundException {
...
-        return getFieldIndexHoles(ColumnFamilyConstants.COLF_RI);
+    public Map<String,Map<String,FieldIndexHole>> getReversedFieldIndexHoles(double minThreshold) throws TableNotFoundException, IOException {
+        return getFieldIndexHoles(ColumnFamilyConstants.COLF_RI, minThreshold);
     }
     
-    /**
-     * Supplies field index hole for {@link #getFieldIndexHoles()} and {@link #getReversedFieldIndexHoles()}.
-     */
-    private Map<String,Map<String,FieldIndexHole>> getFieldIndexHoles(Text indexColumnFamily) throws TableNotFoundException {
+    private Map<String,Map<String,FieldIndexHole>> getFieldIndexHoles(Text targetColumnFamily, double minThreshold) throws TableNotFoundException, IOException {
```
Comment (ivakegg) [inline src/main/java/datawave/query/util/AllFieldMetadataHelper.java:1087]: should we pass in a datatype filter?

Comment (lbschanno) [reply src/main/java/datawave/query/util/AllFieldMetadataHelper.java:1087]: We can add filters for both field names and datatypes easily enough. Are these results cached on a per-query basis or for longer than that?

**Review by ivakegg — APPROVED** [review]

**Pull request merged**